components/list/list-item-generic.js

```diff
@@ -0,0 +1,80 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+export class ListItemGeneric extends RtlMixin(LitElement) {
```

Comment:
I hesitate to chime before this is ready alas, earlier the better I think...  

Currently, I think this "generic" component is simply to define a generic layout.  It is not a list item, rather, it is used by a list-item implementation.  If that remains to be the case, then I think this should be `ListItemGenericLayout` / `d2l-list-item-generic-layout`.  I was mulling over the idea of whether this should actually just be a mixin, and would be happy to hear thoughts on that too.

Besides that, it looks like this class currently does not need to be exported.

Reply:
@m6jones What do you think about this?

Reply:
So, the new name YES! I always hated the name and hoping that someone later would make a better one. In the design doc I avoiding naming it.

So this doesn't need to be a mixin as it can do everything it requires without being one. Which makes it easier to use and edit. So the decision which I mulled over for a while comes down to those key factors. 

The only pro for a mixin was to pre optimize by minimizing the dom depth. Didn't seem worth it. 

Side note @dbatiste, you comments are always welcome before it is ready. earlier the better is good. The goal with draft PR approval from our team then making it a real PR for your approval is so we can save your time when you are more busy as hopefully we get at least all the small things. If you have time to comment on the draft / want to take part it will just make things better.